src/hook/CartContext.tsx

```diff
@@ -1,18 +1,24 @@
-import React, { createContext, useContext, useEffect, useState } from 'react';
+import { createContext, useContext, useEffect, useState, ReactNode } from 'react';
 import { CartItem as CartItemType } from '../types/cartItem';
 
 interface CartContextType {
   productsInCart: CartItemType[];
   addToCart: (item: CartItemType) => void;
   updateQuantity: (id: string, newQuantity: number) => void;
   removeFromCart: (id: string) => void;
-  clearCart: () => void; // Add clearCart to the context type
+  clearCart: () => void;
+  isRemoving: string | null; // Track which item is being removed
```

Comment:
can we use optional here `isRemoving?: string`

Reply:
Fixed